config/aaa.py

```diff
@@ -161,7 +161,8 @@ def passkey(ctx, secret):
 @click.option('-a', '--auth_type', help='Authentication type, default pap', type=click.Choice(["chap", "pap", "mschap", "login"]))
 @click.option('-o', '--port', help='TCP port range is 1 to 65535, default 49', type=click.IntRange(1, 65535), default=49)
 @click.option('-p', '--pri', help="Priority, default 1", type=click.IntRange(1, 64), default=1)
-def add(address, timeout, key, auth_type, port, pri):
+@click.option('-v', '--vrf', help="Management vrf, default is no vrf")
```

Comment:
prefer not to let user to specify the management vrf name, the management vrf is created internally. Here the command line option should be --use-mgmt-vrf. If option is here, then use management vrf for tacacs client. If option is not provided, then do not use management vrf for tacacs client.

Reply:
It would be best to agree on Option 1.

Reply:
option is fine, but we need to make sure mgmt vrf is created before we configure tacacs client.

Reply:
Got it. Can I assume that it would be merged? Accordingly I will modify the code related to parsing the parameter and using it in PAM & NSS and raise pull request for that.

Reply:
The mgmt vrf configuration is outside the scope of tacacs. If "--use-mgmt-vrf" is passed but the bind to device with SO_BINDTODEVICE setsockopt fails, then it shouldn't operate on default vrf.

Reply:
Agreed. I will be making changes in PAM/NSS accordingly to fail the authentication process if user had specified "--use-mgmt-vrf" and if user had NOT configured management vrf interface with fixed name "mgmt". Can I assume that the PR will be merged?

Reply:
@kannankvs, can you help to me to understand following flow. when there is ssh request, the tacacs client will be invoked to establish a new session for authentication. if --use-mgmt-vrf is applied and mgmt vrf is not there, you will fail the request. Then, mgmt vrf is created. another ssh request comes in, will tacacs client succeed this time?

Is this bind to device operation checked everytime for a new authentication or it is only check at the very first time?

Reply:
Yes, tacacs client will try to bind every time when SSH tries to authenticate. Hence, if mgmt vrf is created later and if any new ssh request comes after that, tacacs client will succeed.

Reply:
i think we are in sync now.

Reply:
@lguohan  - I assume you are fine with having "-v" as short form for "--use-mgmt-vrf". 
Ex: "config tacacs add -v 10.11.55.40" is same as "config tacacs add --use-mgmt-vrf 10.11.55.40". In both cases, it will set the data['vrf'] = "mgmt".

Reply:
use -m

-v is for version or verbose.